File: platforms/swift/examples/core/texture/write_with.swift
import FragmentColor
let renderer = Renderer()
let texture = try await renderer.createStorageTexture([64, 32], TextureFormat.Rgba, nil)
let region_bytes = [0u8; 64 * 32 * 4]
let opt = TextureWriteOptions.whole().withBytesPerRow(256).withRowsPerImage(32)
texture.writeWith(region_bytes, opt)

File: platforms/kotlin/examples/core/texture/write_with.kt
import org.fragmentcolor.*
val renderer = Renderer()
val texture = renderer.createStorageTexture([64, 32], TextureFormat.Rgba, null)
val region_bytes = [0u8; 64 * 32 * 4]
val opt = TextureWriteOptions.whole().withBytesPerRow(256).withRowsPerImage(32)
texture.writeWith(region_bytes, opt)